testdata/gapi/onnx/download_onnx_models.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+import sys
+import os
+from shutil import rmtree as rmdir_with_data
+import hashlib
+
+CUR_DIR = os.getcwd()
+# This directory contains result of "git clone"
+CACHE_DIR = CUR_DIR + '/onnx_models_cache/'
+
+class Model:
+    MB = 1024*1024
+    BUFSIZE = 10*MB
+    # This directory contains large ONNX models that will be used
+    MODELS_DIR = CUR_DIR + '/onnx_models/'
+
+    def __init__(self, **kwargs):
+        self.name = kwargs.pop('name')
+        self.filepath = kwargs.pop('filepath')
+        self.sha = kwargs.pop('sha')
+        self.model_path = self.filepath + self.name + '.onnx'
+
+    def __str__(self):
+        return '[Model]: <{}>'.format(self.name + '\n          '+ self.filepath)
+
+    def sys_lfs_call(self):
+        return 'cd onnx_models_cache && git lfs pull --include=/' + str(self.model_path) + ' --exclude="" '
+
+    def verify(self):
+        if not os.path.exists(self.MODELS_DIR + self.model_path):
+            return False
+        if not self.sha:
+            return False
+        print('[Info]: Verify file:')
+        print('        Expect sha: {}'.format(self.sha))
+        sha = hashlib.sha1()
+        try:
+            with open(self.MODELS_DIR + self.filepath + self.name + '.onnx', 'rb') as f:
+                while True:
+                    buf = f.read(self.BUFSIZE)
+                    if not buf:
+                        break
+                    sha.update(buf)
+            print('        Actual sha: {}'.format(sha.hexdigest()))
+            return self.sha == sha.hexdigest()
+        except Exception as e:
+            print('[Excn]: Catch {}'.format(e))
+
+    def create_dir(self):
+        model_dir_path = self.MODELS_DIR + self.filepath
+        if not os.path.exists(model_dir_path):
+            os.makedirs(model_dir_path, exist_ok=True)
```

Comment:
I'm not sure. Do you verify that folder doesn't exist but still pass `exist_ok` parameter? Could probably remove the `if`

Reply:
Removed `exist_ok=True`. We skip this step and print warning that directory already contains model's folder. Step of `makedirs` won't drop exception with `exist_ok=True` if catalog contains this path. But this case impossible because `if` checks this. 
I removed `exist_ok=True` (False default) but it doesn't change behavior. I don't want drop exception on this step (about remove `if`) because directory which exist can receive new model file instead current (if sha check will fail).

Reply:
Got it, thanks